### ISSUE TITLE
False positive on "Avoid unused private fields"

### ISSUE THREAD

**Issue by Etienne-Schmitt** [phpmd/phpmd#887]
- PHPMD version: 2.9.1
- PHP Version: 7.4.18
- Installation type: composer
- Operating System / Distribution & Version: MacOS 11.3

### Current Behavior
I got an phpmd error : TestPhpMD.php:5      Avoid unused private fields such as '$newString'.

### Expected Behavior
I expected no error because the fields is used in a double quotes string.

### Steps To Reproduce:
I created a class with an private string fields witch is used in double quote string like this : 

TestPhpMD.php
```php
<?

class TestPhpMD
{
    private string $newString = "something";

    public function doSomething(): void
    {
        echo "$this->newString else\n";
    }
}

$test = new TestPhpMD();
$test->doSomething;
```
phpmd.xml
```xml
<?xml version="1.0"?>
<ruleset
        name="Random Name"
        xmlns="http://pmd.sf.net/ruleset/1.0.0"
        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
        xsi:schemaLocation="http://pmd.sf.net/ruleset/1.0.0 http://pmd.sf.net/ruleset_xml_schema.xsd"
        xsi:noNamespaceSchemaLocation="http://pmd.sf.net/ruleset_xml_schema.xsd"
>
    <description>Random description</description>

    <rule ref="rulesets/cleancode.xml">
        <exclude name="StaticAccess"/>
    </rule>
    <rule ref="rulesets/codesize.xml"/>
    <rule ref="rulesets/controversial.xml">
        <exclude name="CamelCaseMethodName"/>
    </rule>
    <rule ref="rulesets/design.xml"/>
    <rule ref="rulesets/naming.xml"/>
    <rule ref="rulesets/unusedcode.xml"/>

    <rule ref="rulesets/controversial.xml/CamelCaseMethodName">
        <properties>
            <property name="allow-underscore" value="true"/>
        </properties>
    </rule>
</ruleset>
```

    $ php TestPhpMD.php
    something else
    $ phpmd TestsPhpMD.php text phpd.xml
    ***/TestsPhpMD.php:5      Avoid unused private fields such as '$newString'.

## Checks before submitting
* [ x ] Be sure that there isn't already an issue about this. See: [Issues list](https://github.com/phpmd/phpmd/issues)
* [ x ] Be sure that there isn't already a pull request about this. See: [Pull requests](https://github.com/phpmd/phpmd/pulls)
* [ x ] I have added every step to reproduce the bug.
* [ x ] If possible I added relevant code examples.
* [ x ] This issue is about 1 bug and nothing more.
* [ x ] The issue has a descriptive title. For example:  "JSON rendering failed on Windows for filenames with space".

**Comment by ihr-it-projekt** [comment]
I have same issue.

**Comment by tvbeek** [comment]
@Etienne-Schmitt thanks for the report. I personally shouldn't use the string in the double quote string but I agree it is preferred to get this working.

Do you have the option to create a PR (with a solution or a failing test)?
